### PR TITLE
babe: re-add slot time algorithm, move arrivalTime out of block and into BlockState

### PR DESCRIPTION
**Pull request by noot**
# Changes
- fix slotTime algorithm to work with blocktree updates
- move arrivalTime out of block struct, save it in the BlockState when AddBlock is called
- note: this branches off of #652 

## Tests:
```
go test ./... -short
```

### Issues:
working towards #567 , but computeSlotForBlock isn't correct. will open an issue for that

### CODE REVIEW

File: dot/state/block.go [hunk 1]
```diff
@@ -270,10 +298,19 @@ func (bs *BlockState) SetBlockData(blockData *types.BlockData) error {
 	return err
 }
 
-// AddBlock adds a block to the blocktree and the DB
 func (bs *BlockState) AddBlock(block *types.Block) error {
+	return bs.AddBlockWithArrivalTime(block, uint64(time.Now().Unix()))
+}
+
+// AddBlock adds a block to the blocktree and the DB
```
Comment (thomasmodeneis) [inline dot/state/block.go]: should be AddBlockWithArrivalTime

File: dot/state/block.go [hunk 2]
```diff
@@ -23,13 +23,15 @@ import (
...
 	"sync"
+	"time"
 
 	"github.com/ChainSafe/gossamer/dot/core/types"
 	babetypes "github.com/ChainSafe/gossamer/lib/babe/types"
 	"github.com/ChainSafe/gossamer/lib/blocktree"
 	"github.com/ChainSafe/gossamer/lib/common"
 	"github.com/ChainSafe/gossamer/lib/common/optional"
 	"github.com/ChainSafe/gossamer/lib/database"
+	// log "github.com/ChainSafe/log15"
```
Comment (thomasmodeneis) [inline dot/state/block.go]: can remove comment ?

Comment (noot) [reply dot/state/block.go]: done

File: dot/state/block.go (hunk 1 above)
Comment (noot) [reply dot/state/block.go]: fixed

**Pull request merged**